### PR TITLE
Adding Container Exec Spec

### PR DESCRIPTION
**Pull request by samkreter**
### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by lmazuel** [comment]
Hi @samkreter. Thanks for your contribution! Please fix all the CI issue first :)
Thank you!

**Comment by lmazuel** [comment]
Hi @samkreter , please fix all issues in job called "MODE=model PR_ONLY=true". Thank you

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 1]
```diff
@@ -175,6 +175,12 @@
           },
           "201": {
             "description": "Created - the container group is created.",
+            "headers": {
```
Comment (lmazuel) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: I don't see any reason for this header, long running operation are handled by the x-ms-long-running-operation flag

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 2]
```diff
@@ -314,6 +321,39 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AsyncOperation"
```
Comment (lmazuel) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: Name seems unrelated to semantic of the operation.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 3]
```diff
@@ -314,6 +321,39 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerInstance/locations/{location}/operations/{operationId}": {
+      "get": {
+        "operationId": "AsyncContainerGroupOperation_Get",
```
Comment (lmazuel) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: Name seems unrelated to semantic of the operation.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 4]
```diff
@@ -1030,6 +1120,73 @@
         }
       }
     },
+    "AsyncOperation": {
```
Comment (lmazuel) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: This is not the way we model long running operation at all. Please see for example:
https://github.com/Azure/azure-rest-api-specs/blob/8b34c73ead836efa86806e7da555ade6692cbde4/specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json#L403

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 5]
```diff
@@ -314,6 +321,39 @@
...
+        "description": "Get the usage for a subscription",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/OperationIdParameter"
```
Comment (lmazuel) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: I really don't understand what you'are trying to do here :(
Even the example "ContainerGroupUsage" doesn't have this information

**Review by lmazuel — CHANGES_REQUESTED** [review]

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 1 above)
Comment (zhedahht) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: When we implemented this feature, we followed the following document about Azure Async Operations:
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-async-operations

Comment (lmazuel) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: I'm sure you implemented the server the right way though :)
I'm saying that inside the Swagger, we have a special flag to say this is a ARM async operation and then generators do all this cumbersome work of headers and polling for free.
So you just use that flag, you define your return type with the final resource, and let Autorest deals with the rest.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/8b34c73ead836efa86806e7da555ade6692cbde4/specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json#L403-L462

**Comment by lmazuel** [comment]
Adding @ravbhatnagar for ARM feedback

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 3 above)
Comment (lmazuel) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: Unless there is something I don't get in your service, if this is the operation to follow your previous async operation, you don't need this at all. If you follow ARM guidelines, this operation will be discovered by the "x-ms-long-running" flag.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 5 above)
Comment (zhedahht) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: The example was incorrect. The latest commit fixed this issue.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 3 above)
Comment (ravbhatnagar) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: There is not need to add this to the swagger spec.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 6]
```diff
@@ -360,6 +395,56 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerInstance/containerGroups/{containerGroupName}/containers/{containerName}/exec": {
```
Comment (ravbhatnagar) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: action name exec can be more descriptive. launchExecCommand etc.

**Review by ravbhatnagar — COMMENTED** [review]
Minor notes. Else looks good.

**Comment by ravbhatnagar** [comment]
I dont have any comments which block the ARM sign off. remove the long running API, name change comment is nice to have. @lmazuel FYI.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 6 above)
Comment (robbiezhang) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: `exec` is the Kubernetes convention.

**Comment by lmazuel** [comment]
@samkreter LGTM. I will update the Python package version in your PR, this will retrigger the CI (failure doesn't seem to be on you) and I will merge when it's green. Thanks!

**Review by lmazuel — APPROVED** [review]
